### PR TITLE
feat(sdk): `LinkedChunk` can hold a value for `Gap`s

### PR DESCRIPTION
**Pull request by Hywan**
This patch updates `ChunkContent::Gap` to hold a content `U`. Thus, `Chunk` and LinkedChunk` both get a new generic parameter `U`. Some methods like `new_gap` or `insert_gap_at` take a new `content: U` parameter.

This type `RoomEvents` (that uses `LinkedChunk`) is also updated accordingly.

And because @bnjbvr asked me to do it with Beatbox, I delegate this task to the king of Beatbox: [Rythmind](https://www.youtube.com/watch?v=IX8p4GAfUoU).

### CODE REVIEW

File: crates/matrix-sdk/src/event_cache/linked_chunk.rs [hunk 1]
```diff
@@ -40,20 +40,20 @@ pub enum LinkedChunkError {
 /// of a single one. A chunk has a maximum capacity of `CAPACITY`. Once a chunk
 /// is full, a new chunk is created. Not all chunks are necessarily entirely
 /// full.
-pub struct LinkedChunk<T, const CHUNK_CAPACITY: usize> {
+pub struct LinkedChunk<T, U, const CHUNK_CAPACITY: usize> {
```
Comment (bnjbvr) [inline crates/matrix-sdk/src/event_cache/linked_chunk.rs:43]: Could we give meaningful names to T and U, to make them easy to understand by just reading the code here? Nothing mandates one-letter names for generics in Rust :)

File: crates/matrix-sdk/src/event_cache/linked_chunk.rs [hunk 2]
```diff
@@ -40,20 +40,20 @@ pub enum LinkedChunkError {
...
     length: usize,
     /// The generator of chunk identifiers.
     chunk_identifier_generator: ChunkIdentifierGenerator,
     /// Marker.
-    marker: PhantomData<Box<Chunk<T, CHUNK_CAPACITY>>>,
+    marker: PhantomData<Box<Chunk<T, U, CHUNK_CAPACITY>>>,
 }
 
-impl<T, const C: usize> LinkedChunk<T, C> {
+impl<T, U, const C: usize> LinkedChunk<T, U, C> {
```
Comment (bnjbvr) [inline crates/matrix-sdk/src/event_cache/linked_chunk.rs:56]: and ditto here

**Review by bnjbvr — APPROVED** [review]
Sounds good! Should we also add a way to *read* a gap value, one way or another?

**Comment by Hywan** [comment]
It's possible to read the `Gap` content with the new `Chunk::content` method introduced in https://github.com/matrix-org/matrix-rust-sdk/pull/3230.

I'm going to rename the generic parameter and merge this PR.

**Pull request merged**